crates/proto/src/lib.rs

```diff
@@ -7,6 +8,7 @@ pub mod generated;
 // RE-EXPORTS
 // ================================================================================================
 
+pub use client::{AcceptHeaderInterceptor, RpcClient, connect};
```

Comment:
If we convert `RpcClient` into a newtype as described in one of the above comments, we could export just the `RpcClient` from here (unless we need to export `AcceptHeaderInterceptor` for some other reasons).

Reply:
`AcceptHeaderInterceptor` still needs to be public as it is part of the public type `RpcClient`.

```
private type client::MetadataInterceptor in public interface
can’t leak private type (rustc E0446)
hint: `client::MetadataInterceptor` declared as private
```